src/comm/TcpAcceptor.cc

```diff
@@ -118,7 +118,6 @@ Comm::TcpAcceptor::swanSong()
     }
 
     conn = NULL;
-    AcceptLimiter::Instance().removeDead(this);
```

Comment:
Please keep this de-registration mechanism, as discussed [earlier](https://github.com/squid-cache/squid/pull/857#pullrequestreview-701117114).

Reply:
Any ideas how? The AcceptLimiter is storing N Call's, a subset M of them refer the the TcpAcceptor calling itself "this" in the above line. It was easy when we had pointers to compare. That is no longer accessible.

Reply:
I can repeat what I suggested [earlier](https://github.com/squid-cache/squid/pull/857#issuecomment-875976489): TcpAcceptor should store its registered callback so that it can call removeDead() with it (instead of `this`). Please let me know what aspects of that sketch need clarification.

> a subset M of them refer the the TcpAcceptor

This detail does not really matter much at this high level, but, AFAICT, a given TcpAcceptor cannot have more than one callback registered with AcceptLimiter because TcpAcceptor in a deferred state cannot attempt accepting again and, hence, cannot register a second callback.

Reply:
Done in c5af077